### PR TITLE
Fix plane model matrix

### LINKED ISSUE #8268: Plane entity not rotating correctly
This came up in a [forum post](https://groups.google.com/d/msg/cesium-dev/_S5Z3oaJdvQ/_JEOWflNBQAJ).

[Sandcastle demonstration](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=7Vfhbts2EH4Vwn8sAxpt1067OXawNk3bAEmXNe6aZB4CWjrbxChSICnHbpB3H0lRjqS4S5YEGFBUCWyRvLvv7rsjeV4SiZYUrkCiEeJwhfZB0SzBf7i5oBm54b7gmlAOshmi6wlH5qF8Jt6IFRqgGWEKwnxWAYNIU8EPeUwjooWsCyxILK6UmdUyu50UGYtfc5oQDX5pwm9auxM+4UvjYSoUtVaNj96/fSK1eSO8h2dSJG9hLgFU8FP3RQ93XvX7L7u/hKjfx52dTu9V52WIdjqdDu5Yk9bgIpWfBGMXIEU17g9AYsrnJ1RHCysRFBpCUjAkVL0YS8LVTMhE4UVN8ffMxCK5EQ8K78MybB7cLOOOLhRJMPLHIgYWcJKAkQU6X+gWurYM5TzlicLGD2MPFJaQiCW89l56GeOrE1jv+kTNnEE0Go1QM2WEw+Wq2crXfC7tk+uY0OogJI6Dkpx9nL2B+wqrK0Wog81bXcA6YHRrFstLpWyc2Kngbso/fzwcX56FyGY0vGsppglwZcBV1dzGwouga1RD1P2GAVuHkhJm1JsYt01dzkG1cyuXR2IuLg3vkpE1Tvm8ucWAyDSjHKplvkVgXzC3RQoH7RB/+XA4Pqiq3NRMlKtxUB7dirnt417yLzCbcFsxrL+LYjj/UQzPUQxfv4tiuPhRDE8rhqn4H6+Iqesqrh+RvF7Qt7nr2Y+de/NXYfbN0efnJ9bzpCWJ/ob4oKCvcj3f3tqJvfyPiZZ09aAOYyze0RXE76TheHuPERZWDhijqRI0xl/en/7cL9z0/qkIOOBU0sSYWPp8lvh9C9Nsfnzr3UkhWU562ftBeVSikQGf64VZ7tperLRwReN8Hnc8gS3r4823WqRPQgeejhCllpAQSRN0qygb3+Pd09ttNeG5ea5+r8Z1pRZwFaQ0KoL3vWeqXdWP0J8+Pg0rbY8gdwyiCxTkNdUqThwj7Tpx2317/oINOeVec3PulztkX8I3m2a/Cnf2ZLjVf4E7fzLc+mFw9ufM46HsmVmGsRB/7d5J4tn5xZY8fhQcHgRZg7Ubwd1X9v9fg/PF/3iQ3gNA3G54NIS9tXv3Qdi99hQEB1JGKHJk/04JjyOiNAN7Bo6FYFMij4Fngc+eVbxXyKTYyjXCxlDpNYO9HOxXmqRCapRJFpjWQUNiqtNcXO1pZo4EjSPlzA/bhdIwpktE49GkUfsVPmmgiBGlzMosY+yUfoVJY2/YNvIVNSZczn/LOxIrsujuHeWTGONh2wzvauk8opLFfwA).

Select `Plane Z` and then select between `Heading`, `Pitch`, and `Roll` in the second drop down. Notice how `Heading` looks exactly the same as `None`. For comparison, select `Box` from the first drop down.

I tracked it down to `PlaneGeometryUpdater : createPrimitiveMatrix`. If the function becomes just `return transform;`, the Z-Plane works, but then the X and Y planes become Z-planes too. 

I think the core problem is how the up vector gets set, since it ignores `transform`. If the function is passed two different transforms with the same Z column, such as when `Heading` vs `None` is selected, the resulting matrix will be the same.

Any thoughts @hpinkos, @OmarShehata?

### PR DESCRIPTION
**Pull request by baothientran**
Fixes #8268 

Currently the rotation is constructed manually to align the plane with the surface of the ellipsoid. This works correctly if the plane's normal is not aligned with the main axis. Instead of transforming the plane's origin and normal to the world coordinate first, we can orient the plane in the local coordinate using its original normal and origin. Then, we multiply the local transform with the world transform to convert to the world coordinate. Because the world transform already has the rotation matrix that will align the plane with the ellipsoid's surface, this should work with all of the examples as before.

@lilleyse or @hpinkos Can you please take a look at it? Please let me know if it is a right approach

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @baothientran!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/plane-model-matrix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by lilleyse** [comment]
@IanLilleyT can you review? I think you had the same idea when you wrote up https://github.com/CesiumGS/cesium/issues/8268

**Comment by lilleyse** [comment]
@baothientran can you update `CHANGES.md`?

**Comment by baothientran** [comment]
`CHANGES.md` is updated

File: Source/DataSources/PlaneGeometryUpdater.js [hunk 1]
```diff
@@ -344,27 +334,19 @@ function createPrimitiveMatrix(
...
     CesiumMath.equalsEpsilon(
-      Math.abs(Cartesian3.dot(up, transformedNormal)),
+      Math.abs(Cartesian3.dot(up, normal)),
       1.0,
       CesiumMath.EPSILON8
     )
   ) {
-    up = Cartesian3.clone(Cartesian3.UNIT_Z, up);
+    up = Cartesian3.clone(Cartesian3.UNIT_Y, up);
     if (
```
Comment (IanLilleyT) [inline Source/DataSources/PlaneGeometryUpdater.js:347]: I know this isn't code you added, but I think we can safely delete this section. There's no way the initial up guess is aligned to both the Z and Y axes, so setting to `UNIT_X` will never happen.

**Comment by baothientran** [comment]
@IanLilleyT I updated to remove the unreachable condition that aligns up with X-axis.

**Comment by IanLilleyT** [comment]
@baothientran thanks! Can you edit the CHANGES.md to be for the october release.

**Comment by IanLilleyT** [comment]
Thanks @baothientran !

**Pull request merged**